src/java/org/streamingpool/core/service/streamfactory/FlattenedStreamFactory.java

```diff
@@ -52,6 +52,7 @@
         StreamId<Iterable<T>> source = filteredId.sourceStreamId();
 
         return Optional.of(Flowable.fromPublisher(discoveryService.discover(source))
+                .filter(Objects::nonNull)
```

Comment:
As far as I know, rxjava 2.0 does not support nulls in flowables. Therefore, this filtering should be unnecessary (or it would throw already before).